### PR TITLE
Chime Hack: Allow Chime to migrate Insatnce_IDs.

### PR DESCRIPTION
**Pull request by ankitaj224**

### CODE REVIEW

File: firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java [hunk 1]
```diff
@@ -363,13 +364,14 @@ private PersistedInstallationEntry getPrefsWithGeneratedIdMultiProcessSafe() {
...
+    // Check if this firebase app is the default (first initialized) instance or is a chime app
+    if (!firebaseApp.getName().equals(CHIME_FIREBASE_APP_NAME)
+        && (!firebaseApp.isDefaultApp() || !prefs.shouldAttemptMigration())) {
       return fidGenerator.createRandomFid();
     }
-    // For a default firebase installation, read the existing iid from shared prefs
+    // For a default/chime firebase installation, read the existing iid from shared prefs
     String fid = iidStore.readIid();
-    if (fid == null) {
+    if (TextUtils.isEmpty(fid)) {
```
Comment (andirayo) [inline firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java]: shouldn't this be a separate PR?

File: firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java [hunk 2]
```diff
@@ -363,13 +364,14 @@ private PersistedInstallationEntry getPrefsWithGeneratedIdMultiProcessSafe() {
   }
 
   private String readExistingIidOrCreateFid(PersistedInstallationEntry prefs) {
-    // Check if this firebase app is the default (first initialized) instance
-    if (!firebaseApp.isDefaultApp() || !prefs.shouldAttemptMigration()) {
+    // Check if this firebase app is the default (first initialized) instance or is a chime app
+    if (!firebaseApp.getName().equals(CHIME_FIREBASE_APP_NAME)
+        && (!firebaseApp.isDefaultApp() || !prefs.shouldAttemptMigration())) {
```
Comment (andirayo) [inline firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java]: Not sure if the bracketing is right here...
I don't know what #shouldAttemptMigration does (so I might be wrong here)...
But it seems, that this should be changed to:

    if ((!firebaseApp.getName().equals(CHIME_FIREBASE_APP_NAME) && !firebaseApp.isDefaultApp())
       || !prefs.shouldAttemptMigration()) {

Comment (ankitaj224) [reply firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java]: Done.

**Comment by ankitaj224** [comment]
> 2 comments

Addressed.

File: firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java (hunk 1 above)
Comment (ankitaj224) [reply firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java]: Small change and seemed relevant as the change the same method. Will do it next time.
